content/content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java

```diff
@@ -1144,7 +1145,14 @@ public void doAttachitem(RunData data)
 
 		//toolSession.setAttribute(STATE_LIST_SELECTIONS, new TreeSet());
 
-		String itemId = params.getString("itemId");
+		String itemId = null;
+		try {
+		    itemId = URLDecoder.decode(params.getString("itemId"), "UTF-8");
+		} catch (java.io.UnsupportedEncodingException e) {
+		    // should be impossible. use original, with warning
+		    logger.warn("UTF-8 unsupported???");
```

Comment:
UTF-8 is guaranteed to be available. https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html
It's just a shame there's not a `URLDecoder` method that accepts a `Charset`.